### ISSUE TITLE
e2e-test: Check if the pod is really running before execing into it

### ISSUE THREAD

**Issue by surajssd** [confidential-containers/cloud-api-adaptor#1492]
Right now we see this field called `podState` and decide if we should exec into this pod. But `podState` may not be the current state of the pod it is a field used to define the intent from the test writer. But in some places we use it as if it is the current state.

https://github.com/confidential-containers/cloud-api-adaptor/blob/cc34a129e28fba08533481c749c82f7d87f0d017/test/e2e/common_suite_test.go#L270

So before we exec into a pod we should check if it is really running? It is possible that the pod is in `CrashLoopBackOff` state this is the phase generally where the pod starts, reports as `Running` and then fails for whatever reason and then goes into the `CrashLoopBackOff`.

https://github.com/confidential-containers/cloud-api-adaptor/blob/cc34a129e28fba08533481c749c82f7d87f0d017/test/e2e/common_suite_test.go#L282

**Comment by mkulke** [comment]
Can we fix that by adding a health check? so it'll be never reported as running?

**Comment by surajssd** [comment]
> Can we fix that by adding a health check to the pod spec? so it'll be never reported as running?

Yes, healthchecks could help with the nginx pod, but not with regular busybox pods.
